### PR TITLE
Maya: Use custom staging dir function for Maya renders - OP-5265

### PR DESCRIPTION
**Pull request by tokejepsen**
## Changelog Description
Check for custom staging dir when setting the renders output folder in Maya.

## Testing notes:
1. Setup custom staging dir; https://openpype.io/docs/project_settings/settings_project_global#custom-staging-directory-profiles
2. Setup render in Maya.
3. Publish and validate renders go to the custom staging dir.

### CODE REVIEW

File: openpype/hosts/maya/plugins/publish/validate_render_image_rule.py [hunk 1]
```diff
@@ -37,15 +43,26 @@ def process(self, instance):
     @classmethod
     def repair(cls, instance):
 
-        required_images_rule = cls.get_default_render_image_folder(instance)
+        required_images_rule = cls.get_default_render_image_folder(
+            cls, instance
+        )
```
Comment (moonyuet) [inline openpype/hosts/maya/plugins/publish/validate_render_image_rule.py:48]:
```suggestion
        required_images_rule = cls.get_default_render_image_folder(
        instance
        )
```
it might error out with (cls, instance)

File: openpype/hosts/maya/plugins/publish/validate_render_image_rule.py [hunk 2]
```diff
@@ -24,8 +26,12 @@ class ValidateRenderImageRule(pyblish.api.InstancePlugin):
 
     def process(self, instance):
 
-        required_images_rule = self.get_default_render_image_folder(instance)
-        current_images_rule = cmds.workspace(fileRuleEntry="images")
+        required_images_rule = os.path.normpath(
+            self.get_default_render_image_folder(self, instance)
```
Comment (moonyuet) [inline openpype/hosts/maya/plugins/publish/validate_render_image_rule.py:30]:
```suggestion
            self.get_default_render_image_folder(instance)
```
it may error out with (self, instance).

File: openpype/hosts/maya/plugins/publish/validate_render_image_rule.py (hunk 1 above)
Comment (tokejepsen) [reply openpype/hosts/maya/plugins/publish/validate_render_image_rule.py:48]: Why do you think it'll error?

Comment (moonyuet) [reply openpype/hosts/maya/plugins/publish/validate_render_image_rule.py:48]: it will error with TypeError for the function missing one required positional argument. `cls.get_default_render_image_folder` expects 1 argument instead of 2.

Comment (tokejepsen) [reply openpype/hosts/maya/plugins/publish/validate_render_image_rule.py:48]: But the function arguments is changed?

Comment (moonyuet) [reply openpype/hosts/maya/plugins/publish/validate_render_image_rule.py:48]: maybe I am wrong as I notice the` @staticmethod`

Comment (tokejepsen) [reply openpype/hosts/maya/plugins/publish/validate_render_image_rule.py:48]: The code seems to be working here. We could remove the static method decorator?
Not sure why its there anyways.

Comment (moonyuet) [reply openpype/hosts/maya/plugins/publish/validate_render_image_rule.py:48]: yes, we should remove @staticmethod as we dont need it for this case.

Comment (iLLiCiTiT) [reply openpype/hosts/maya/plugins/publish/validate_render_image_rule.py:48]: It is static method, but we should remove `cls` attribute from `get_default_render_image_folder`.

Comment (tokejepsen) [reply openpype/hosts/maya/plugins/publish/validate_render_image_rule.py:48]: Ok, lets take this back a bit. I need to log a message for the user as debug call. What would be the best route here?

- pass in the log to `get_default_render_image_folder`
- remove the static method decorator and turn `get_default_render_image_folder` into a class method? Why is `get_default_render_image_folder` a static method to begin with?
- alternatives?

Comment (tokejepsen) [reply openpype/hosts/maya/plugins/publish/validate_render_image_rule.py:48]: Following @BigRoy advice. Please unresolve the conversation if you think its not the right direction.

**Review by LiborBatek — COMMENTED** [review]
I have setup the custom staging dir via OP Settings (see imgs).
image

Get following error during ```Publish``` of Render instance to DL.

image

**Comment by tokejepsen** [comment]
@LiborBatek sorry. Should be fixed now.

**Review by moonyuet — APPROVED** [review]
haven't tested, code looks good to me

**Review by Minkiu — APPROVED** [review]

File: openpype/hosts/maya/plugins/publish/extract_pointcache.py [hunk 3]
```diff
@@ -108,7 +108,8 @@ def process(self, instance):
         }
         instance.data["representations"].append(representation)
 
-        instance.context.data["cleanupFullPaths"].append(path)
+        if not instance.data.get("stagingDir_persistent", False):
```
Comment (iLLiCiTiT) [inline openpype/hosts/maya/plugins/publish/extract_pointcache.py:111]: Why we do combine camel and snake case? `stagingDirIsPersistent` or `stagingDirPersist`

File: openpype/hosts/maya/plugins/publish/extract_thumbnail.py [hunk 4]
```diff
@@ -92,7 +92,8 @@ def process(self, instance):
             "Create temp directory {} for thumbnail".format(dst_staging)
         )
         # Store new staging to cleanup paths
-        instance.context.data["cleanupFullPaths"].append(dst_staging)
+        if not instance.data.get("stagingDir_persistent", False):
```
Comment (iLLiCiTiT) [inline openpype/hosts/maya/plugins/publish/extract_thumbnail.py:95]: Why this? It is in temp directory, it will keep a lot of files for windows users...

File: openpype/hosts/maya/plugins/publish/extract_pointcache.py (hunk 3 above)
Comment (tokejepsen) [reply openpype/hosts/maya/plugins/publish/extract_pointcache.py:111]: This is a hiccup from the original implementation: https://github.com/ynput/OpenPype/pull/4583

Do you want it changed?

File: openpype/hosts/maya/plugins/publish/extract_thumbnail.py (hunk 4 above)
Comment (tokejepsen) [reply openpype/hosts/maya/plugins/publish/extract_thumbnail.py:95]: This is part of the feature design; https://openpype.io/docs/project_settings/settings_project_global#custom-staging-directory-profiles

If a user enables persistent staging, it will be their responsibility to clean up any files.

File: openpype/hosts/maya/plugins/publish/extract_pointcache.py (hunk 3 above)
Comment (iLLiCiTiT) [reply openpype/hosts/maya/plugins/publish/extract_pointcache.py:111]: No.

File: openpype/hosts/maya/plugins/publish/extract_thumbnail.py (hunk 4 above)
Comment (iLLiCiTiT) [reply openpype/hosts/maya/plugins/publish/extract_thumbnail.py:95]: But this is not in custom staging dir. This is in tempfolder...

Comment (tokejepsen) [reply openpype/hosts/maya/plugins/publish/extract_thumbnail.py]: Yeah, good point. Have removed.

**Comment by tokejepsen** [comment]
@LiborBatek could you give this another try?

**Review by LiborBatek — CHANGES_REQUESTED** [review]
Did set my ```transient```folder in OP Settings first:
image

also here:
image


But getting this error during publishing of render:

image

**Comment by tokejepsen** [comment]
@LiborBatek you should be able to "repair" that validation?

**Comment by LiborBatek** [comment]
> @LiborBatek you should be able to "repair" that validation?

Will try that...no prob. I wasnt sure it should be popping up so I rather mentioned it.

**Comment by LiborBatek** [comment]
I did new test with different workfile / scene and it successfully published render to DL.

it created folder for rendering but even during renders it was empty with just JSON file:

image

the actual folder for rendered image was:
image

DL ```PublishJob``` failed then due to missing files
image

**Comment by LiborBatek** [comment]
Did another go with slightly different ```transient``` folder defined to examine it more clearly

it uses the path and create the folders/path for rendering but again being it empty without any rendered frames during the ```render job``` on DL

image

**Comment by tokejepsen** [comment]
@LiborBatek should be fixed now. Give it another try please.

**Review by LiborBatek — APPROVED** [review]
Now it successfully render to ```staging directory``` during render job on DL.

Note: it removes rendered images even when ```persistent files``` being ```on```.

**Review by antirotor — CHANGES_REQUESTED** [review]
Actually, we should fix the cleanup of rendered images when __persistent files__ are on.

**Comment by tokejepsen** [comment]
@LiborBatek can you try this again?
The files should persist now.

**Review by LiborBatek — COMMENTED** [review]
It correctly uses the ```transient folder``` but I got Publish Job errors on DL and after first run of publish job my rendered frames are gone.

I have set the ```Persistent``` ```On``` tho.

Not sure why DL is failing on ```Publish Job``` and if connected somehow... 

```
2023-08-23 09:47:09:  0: STDOUT: C:\TEMP\pyblish_tmp_3k0c05j5
2023-08-23 09:47:09:  0: STDOUT: ERROR:pyblish.plugin:Traceback (most recent call last):
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Program Files (x86)\OpenPype\3.16.0\dependencies\pyblish\plugin.py", line 527, in __explicit_process
2023-08-23 09:47:09:  0: STDOUT:     runner(*args)
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\cleanup_explicit.py", line 29, in process
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\cleanup_explicit.py", line 145, in _remove_empty_dirs
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 748, in rmtree
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 621, in _rmtree_unsafe
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 621, in _rmtree_unsafe
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 621, in _rmtree_unsafe
2023-08-23 09:47:09:  0: STDOUT:   [Previous line repeated 1 more time]
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 630, in _rmtree_unsafe
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 628, in _rmtree_unsafe
2023-08-23 09:47:09:  0: STDOUT: PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'D:\\Projects\\Libor_Tests\\tmp_render\\assets\\props\\screwdriver_B\\work\\renderlayer\\renderLookdevMain\\220901_screwdriver_B_workfileLookdev_v003\\Main\\Main\\maya\\220901_screwdriver_B_workfileLookdev_v003\\Main'
2023-08-23 09:47:09:  0: STDOUT: Traceback (most recent call last):
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Program Files (x86)\OpenPype\3.16.0\dependencies\pyblish\plugin.py", line 527, in __explicit_process
2023-08-23 09:47:09:  0: STDOUT:     runner(*args)
2023-08-23 09:47:09:  0: STDOUT:   File "<string>", line 29, in process
2023-08-23 09:47:09:  0: STDOUT:   File "<string>", line 145, in _remove_empty_dirs
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 748, in rmtree
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 621, in _rmtree_unsafe
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 621, in _rmtree_unsafe
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 621, in _rmtree_unsafe
2023-08-23 09:47:09:  0: STDOUT:   [Previous line repeated 1 more time]
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 630, in _rmtree_unsafe
2023-08-23 09:47:09:  0: STDOUT:   File "C:\Users\qam\.pyenv\pyenv-win\versions\3.9.7\lib\shutil.py", line 628, in _rmtree_unsafe
2023-08-23 09:47:09:  0: STDOUT: PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'D:\\Projects\\Libor_Tests\\tmp_render\\assets\\props\\screwdriver_B\\work\\renderlayer\\renderLookdevMain\\220901_screwdriver_B_workfileLookdev_v003\\Main\\Main\\maya\\220901_screwdriver_B_workfileLookdev_v003\\Main'
2023-08-23 09:47:09:  0: STDOUT: !!! ERR: 2023-08-23 09:47:09,385 >>> { CLI-publish }: [ Failed ExplicitCleanUp: [WinError 32] The process cannot access the file because it is being used by another process: 'D:\\Projects\\Libor_Tests\\tmp_render\\assets\\props\\screwdriver_B\\work\\renderlayer\\renderLookdevMain\\220901_screwdriver_B_workfileLookdev_v003\\Main\\Main\\maya\\220901_screwdriver_B_workfileLookdev_v003\\Main' -- ('C:\\Program Files (x86)\\OpenPype\\3.16.0\\openpype\\plugins\\publish\\cleanup_explicit.py', 628, '_rmtree_unsafe', '') ] 
2023-08-23 09:47:09:  0: STDOUT: >>> run disk mapping command ...
2023-08-23 09:47:09:  0: STDOUT: *** disk mapping arguments: ['subst', 'P:\\', 'D:']
2023-08-23 09:47:09:  0: STDOUT: >>> Logging to server is turned ON
2023-08-23 09:47:09:  0: STDOUT: >>> Finding latest version with [ 3.16.0 ]
2023-08-23 09:47:09:  0: STDOUT:   - booting version: 3.16.0
2023-08-23 09:47:09:  0: STDOUT: >>> Injecting OpenPype version to running environment  ...
2023-08-23 09:47:09:  0: STDOUT: >>> Validating version in frozen [ C:\Program Files (x86)\OpenPype\3.16.0 ]
2023-08-23 09:47:09:  0: STDOUT: --- version is valid
2023-08-23 09:47:09:  0: STDOUT: >>> loading environments ...
2023-08-23 09:47:09:  0: STDOUT:   - for Avalon ...
2023-08-23 09:47:09:  0: STDOUT:   - global OpenPype ...
2023-08-23 09:47:09:  0: STDOUT:   - for modules ...
2023-08-23 09:47:09:  0: STDOUT: *** OpenPype [3.16.0] 
2023-08-23 09:47:09:  0: STDOUT: >>> Using OpenPype from [ C:\Program Files (x86)\OpenPype\3.16.0 ]
2023-08-23 09:47:09:  0: STDOUT: ... OpenPype variant:      [ production ]
2023-08-23 09:47:09:  0: STDOUT: ... Running OpenPype from: [ C:\Program Files (x86)\OpenPype\3.16.0 ]
2023-08-23 09:47:09:  0: STDOUT: ... Using mongodb:         [ localhost ]
2023-08-23 09:47:09:  0: STDOUT: ... Logging to MongoDB:    [ localhost ]
2023-08-23 09:47:09:  0: STDOUT: ...   - port:              [ 27017 ]
2023-08-23 09:47:09:  0: STDOUT: ...   - database:          [ openpype ]
2023-08-23 09:47:09:  0: STDOUT: ...   - collection:        [ logs ]
2023-08-23 09:47:09:  0: STDOUT: ...   - user:              [ <N/A> ]
2023-08-23 09:47:09:  0: STDOUT: setting target: deadline
2023-08-23 09:47:09:  0: STDOUT: setting target: farm
2023-08-23 09:47:09:  0: STDOUT: Using plugins:
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_cleanup_keys.py.CollectCleanupKeys'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_current_context.py.CollectCurrentContext'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_current_shell_file.py.CollectCurrentShellFile'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_datetime_data.py.CollectDateTimeData'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_from_create_context.py.CollectFromCreateContext'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_host_name.py.CollectHostName'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_machine_name.py.CollectMachineName'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_modules.py.CollectModules'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_shell_workspace.py.CollectShellWorkspace'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_time.py.CollectTime'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_settings.py.CollectSettings'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_anatomy_object.py.CollectAnatomyObject'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_rendered_files.py.CollectRenderedFiles'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_context_entities.py.CollectContextEntities'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\dependencies\pyblish\plugins\collect_current_date.py.CollectCurrentDate'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\dependencies\pyblish\plugins\collect_current_user.py.CollectCurrentUser'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\dependencies\pyblish\plugins\collect_current_working_directory.py.CollectCurrentWorkingDirectory'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\repair_unicode_strings.py.RepairUnicodeStrings'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_current_pype_user.py.CollectCurrentUserPype'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_anatomy_context_data.py.CollectAnatomyContextData'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\modules\deadline\plugins\publish\collect_default_deadline_server.py.CollectDefaultDeadlineServer'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_context_label.py.CollectContextLabel'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\modules\deadline\plugins\publish\collect_pools.py.CollectDeadlinePools'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_anatomy_instance_data.py.CollectAnatomyInstanceData'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_resources_path.py.CollectResourcesPath'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_source_for_source.py.CollectSourceForSource'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_custom_staging_dir.py.CollectCustomStagingDir'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_input_representations_to_versions.py.CollectInputRepresentationsToVersions'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\modules\deadline\plugins\publish\collect_publishable_instances.py.CollectDeadlinePublishableInstances'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\validate_asset_docs.py.ValidateAssetDocs'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\validate_filesequences.py.ValidateFileSequences'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\validate_unique_subsets.py.ValidateSubsetUniqueness'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\modules\deadline\plugins\publish\validate_deadline_pools.py.ValidateDeadlinePools'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\modules\deadline\plugins\publish\validate_expected_and_rendered_files.py.ValidateExpectedFiles'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\validate_publish_dir.py.ValidatePublishDir'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\validate_resources.py.ValidateResources'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\collect_comment.py.CollectComment'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\extract_hierarchy_avalon.py.ExtractHierarchyToAvalon'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\extract_hierarchy_to_ayon.py.ExtractHierarchyToAYON'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\extract_thumbnail_from_source.py.ExtractThumbnailFromSource'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\extract_scanline_exr.py.ExtractScanlineExr'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\extract_thumbnail.py.ExtractThumbnail'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\extract_color_transcode.py.ExtractOIIOTranscode'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\extract_review.py.ExtractReview'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\extract_burnin.py.ExtractBurnin'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\extract_review_slate.py.ExtractReviewSlate'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\extract_colorspace_data.py.ExtractColorspaceData'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\integrate_subset_group.py.IntegrateSubsetGroup'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\preintegrate_thumbnail_representation.py.PreIntegrateThumbnails'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\integrate_resources_path.py.IntegrateResourcesPath'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\integrate.py.IntegrateAsset'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\integrate_thumbnail.py.IntegrateThumbnails'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\integrate_thumbnail_ayon.py.IntegrateThumbnailsAYON'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\integrate_hero_version.py.IntegrateHeroVersion'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\integrate_inputlinks.py.IntegrateInputLinks'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\cleanup.py.CleanUp'>
2023-08-23 09:47:09:  0: STDOUT: <class 'C:\Program Files (x86)\OpenPype\3.16.0\openpype\plugins\publish\cleanup_explicit.py.ExplicitCleanUp'>
2023-08-23 09:47:09:  0: STDOUT: [WinError 32] The process cannot access the file because it is being used by another process: 'D:\\Projects\\Libor_Tests\\tmp_render\\assets\\props\\screwdriver_B\\work\\renderlayer\\renderLookdevMain\\220901_screwdriver_B_workfileLookdev_v003\\Main\\Main\\maya\\220901_screwdriver_B_workfileLookdev_v003\\Main'
2023-08-23 09:47:10:  0: INFO: Process exit code: 1
2023-08-23 09:47:10:  0: Done executing plugin command of type 'Render Task'

**Comment by tokejepsen** [comment]
@LiborBatek sorry I cant replicate your issue. It all seems to work here.

Could you maybe give it another try now?

**Review by LiborBatek — APPROVED** [review]
Ok so I have finally successfully tested this PR and all working as supposed to!

When using ```transient``` folder it correctly uses it for rendering and if not ```Persistent``` enabled it clears all the rendered frames too.

All working fine!

**Review by antirotor — APPROVED** [review]

File: openpype/modules/deadline/plugins/publish/submit_publish_job.py
Comment (BigRoy) [inline openpype/modules/deadline/plugins/publish/submit_publish_job.py:1]: Saw this PR got merged just now - but can someone remind me why the changes of this particular file in this PR are part of this PR?

This seems wrong?

Comment (tokejepsen) [reply openpype/modules/deadline/plugins/publish/submit_publish_job.py:1]: Think this is a weird diff cause I didn't change do all these changes.
